response/slack/client.py

```diff
@@ -79,10 +79,20 @@ def get_channel_id(self, name):
 
             for channel in response["channels"]:
                 if channel["name"] == name:
+                    if channel['is_archived'] and auto_unarchive:
+                        self.unarchive_channel(channel['id'])
                     return channel["id"]
 
         raise SlackError(f"Channel '{name}' not found")
 
+    def unarchive_channel(self, channel_id):
```

Comment:
There is already an unarchive_channel function on line 143, which will override this